src/index.js

```diff
@@ -0,0 +1,229 @@
+  let degC = 0.0;
+  let landscape = landscapes.neutral;
+
+  if (state.tempUnits === '°F') {
+    degC = convertFtoC(state.temp);
+  } else {
+    degC = state.temp;
+  }
+  if (degC > 30) {
+    landscape = landscapes.hot;
+    color = 'rgba(150, 90, 30, 0.5)';
+  } else if (degC > 25) {
+    landscape = landscapes.warm;
+    color = 'rgba(120, 60, 40, 0.5)';
+  } else if (degC > 15) {
+    landscape = landscapes.neutral;
+    color = 'rgba(100, 110, 10, 0.5)';
+  } else if (degC > 0) {
+    landscape = landscapes.cool;
+    color = 'rgba(80, 10, 110, 0.5)';
+  } else if (degC <= 0) {
+    landscape = landscapes.cold;
+    color = 'rgba(5, 5, 110, 0.5)';
+  }
+
+  let backgroundColors = `radial-gradient(${color}, rgba(0, 0, 0, 0))`;
+
+  const backgroundColor = document.getElementById('weather_body');
+  backgroundColor.style.backgroundImage = backgroundColors;
+
+  const landscapeReturn = document.getElementById('landscape');
+  landscapeReturn.textContent = landscape;
+
+  const tempContainer = document.querySelector('#numeric_temperature');
+  tempContainer.textContent = `${state.temp}`;
+
+  const unitsContainer = document.querySelector('#temperature_units');
+  unitsContainer.textContent = `${state.tempUnits}`;
+};
+
+const changeUnits = () => {
+  const currentUnits = state.tempUnits;
+  // document.getElementById('#temperature_units').textContent;
+  // state.tempUnits = currentUnits;
+  console.log('current units: ' + currentUnits);
+  if (currentUnits === '°C') {
+    state.temp = convertCtoF(state.temp);
+    state.tempUnits = '°F';
+  } else if (currentUnits === '°F') {
+    state.temp = convertFtoC(state.temp);
+    state.tempUnits = '°C';
+  }
+  changeLandscapeTemp();
+};
+
+const increaseTemp = () => {
+  state.temp += 1;
+  changeLandscapeTemp();
+};
+
+const reset = () => {
+  state.temp = 25;
+  state.tempUnits = '°C';
+  changeLandscapeTemp();
+  resetLocation();
+};
+
+const decreaseTemp = () => {
+  state.temp -= 1;
+  changeLandscapeTemp();
+};
+
+const searchLocation = () => {
+  axios
+    .get('http://localhost:5000/location', {
+      params: {
+        q: state.city,
+      },
+    })
+    .then((response) => {
+      console.log('success!' + JSON.stringify(response.data[0]));
+      state.lat = response.data[0].lat;
+      state.lon = response.data[0].lon;
+      searchTemperature();
+    })
+    .catch((error) => {
+      console.log('searchLocation error: ' + error.response);
+    });
+};
+
+const searchTemperature = () => {
+  axios
+    .get('http://localhost:5000/weather', {
+      params: {
+        lat: state.lat,
+        lon: state.lon,
+      },
+    })
+    .then((response) => {
+      console.log('success!' + JSON.stringify(response.data.current.temp));
+      state.temp = convertKtoC(response.data.current.temp);
+      changeLandscapeTemp();
+    })
+    .catch((error) => {
+      console.log('searchTemperature error: ' + error.response);
+    });
+};
+
+const changeCity = () => {
```

Comment:
When I am typing in the input it should change the city name. From wave 3: 
- A text input element that allows the user to change the city name
You should have an event listener for input like
```javascript
 const cityNameInput = document.getElementById('cityNameInput');
  cityNameInput.addEventListener('input', updateCityName);
```